### PR TITLE
Introduce `DYNAMODB2` version store type, deprecate `DYNAMODB` version store type

### PR DESCRIPTION
**Pull request by snazy**
The current `DYNAMODB` version store type uses a lot of attributes in the `objs` collection, which cause quite some overhead in storage. The new `DYNAMODB2` version store type works like `DYNAMODB`, but uses a single attributes for all object values.

### CODE REVIEW

File: servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/storage/DynamoDB2BackendBuilder.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.projectnessie.quarkus.config.QuarkusDynamoDBConfig;
+import org.projectnessie.quarkus.providers.versionstore.StoreType;
+import org.projectnessie.versioned.storage.common.persist.Backend;
+import org.projectnessie.versioned.storage.dynamodb2.DynamoDB2BackendConfig;
+import org.projectnessie.versioned.storage.dynamodb2.DynamoDB2BackendFactory;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
+@StoreType(DYNAMODB2)
+@Dependent
+public class DynamoDB2BackendBuilder implements BackendBuilder {
```
Comment (dimas-b) [inline servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/storage/DynamoDB2BackendBuilder.java:31]: `Dynamo` and `DB2` is one class name ;)

File: versioned/storage/dynamodb2/src/main/java/org/projectnessie/versioned/storage/dynamodb2/DynamoDB2BackendFactory.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.dynamodb2;
+
+import jakarta.annotation.Nonnull;
+import org.projectnessie.versioned.storage.common.persist.BackendFactory;
+
+public class DynamoDB2BackendFactory implements BackendFactory<DynamoDB2BackendConfig> {
+
+  public static final String NAME = "DynamoDB";
```
Comment (dimas-b) [inline versioned/storage/dynamodb2/src/main/java/org/projectnessie/versioned/storage/dynamodb2/DynamoDB2BackendFactory.java:23]: `DynamoDB2`?

**Review by dimas-b — COMMENTED** [review]
LGTM with a minor name comment

File: servers/quarkus-config/src/main/java/org/projectnessie/quarkus/config/VersionStoreConfig.java [hunk 3]
```diff
@@ -28,7 +28,11 @@ public interface VersionStoreConfig {
   enum VersionStoreType {
     IN_MEMORY,
     ROCKSDB,
+    /** JDBC variant using many distinct attributes. */
+    @Deprecated
     DYNAMODB,
+    /** JDBC variant using many distinct attributes. */
```
Comment (adutra) [inline servers/quarkus-config/src/main/java/org/projectnessie/quarkus/config/VersionStoreConfig.java:34]: Wrong comment?

Comment (snazy) [reply servers/quarkus-config/src/main/java/org/projectnessie/quarkus/config/VersionStoreConfig.java:34]: Rebase mess :(

**Review by dimas-b — APPROVED** [review]

**Pull request merged**